data/src/main/java/com/knichu/data/datasource/CityLocationDataSource.kt

```diff
@@ -0,0 +1,17 @@
+package com.knichu.data.datasource
+
+import com.knichu.data.dto.response.CityLocationResponseDTO
+import com.knichu.data.service.CityLocationService
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.schedulers.Schedulers
+import javax.inject.Inject
+
+class CityLocationDataSource @Inject constructor(
+    private val cityLocationService: CityLocationService
+) : BaseNetworkDataSource() {
+    fun getAllCityLocation(): Single<CityLocationResponseDTO> {
+        return cityLocationService.getAllCityLocation()
+            .subscribeOn(Schedulers.io())
+            .map { checkResponse(it) }
```

Comment:
.subscribeOn(Schedulers.io())
```suggestion
            .map { checkResponse(it) }
            .subscribeOn(Schedulers.io())
```